tools/PI/DevHome.PI/ViewModels/BarWindowViewModel.cs

```diff
@@ -236,4 +253,35 @@ private void PerfCounterHelper_PropertyChanged(object? sender, PropertyChangedEv
             });
         }
     }
+
+    public void ExternalToolButton_Click(object sender, RoutedEventArgs e)
+    {
+        if (sender is Button clickedButton)
+        {
+            if (clickedButton.Tag is ExternalTool tool)
+            {
+                InvokeTool(tool, TargetAppData.Instance.TargetProcess?.Id, TargetAppData.Instance.HWnd);
+            }
+        }
+    }
+
+    public void ManageExternalToolsButton_ExternalToolLaunchRequest(object sender, ExternalTool tool)
+    {
+        InvokeTool(tool, TargetAppData.Instance.TargetProcess?.Id, TargetAppData.Instance.HWnd);
+    }
+
+    private void InvokeTool(ExternalTool tool, int? id, HWND hWnd)
+    {
+        var process = tool.Invoke(id, hWnd);
```

Comment:
will tool.Invoke throw if id is null?

Reply:
Nope. This tells the external tool that we don't have a Process ID to pass to the tool. In this case, "0" and "null" have 2 different meanings.

Reply:
I changed the param name to pid to make it more clear what sort of ID this is.